### DISCUSSION TITLE
Any interest in a WebView2 window interface?

### DISCUSSION THREAD

**Discussion by WarlockD** [space-wizards/space-station-14#4286] [Questions] [answered]
I did some pr's in TG (paper, device networking that no one uses bah) and currently working on a WebView2 web interface (in C++ but it applies to C# as well) to test out tgui web windows without having to load up an instance of tg each time.

I have gotten fairly familiar with the interface and the glue between that and your host is either binding a javascript object or a simple command to pass a string back and forth and decode it as a json object.  Mind you control the entire web window and can prevent any external access if you so want or even catch urls.  You can even configure it without the need for a temp folder, having the client transfer images, css and code in a side channel.  Since it runs Edge Chromium you can do some nifty things with it.  I am also familiar with using Chromium Embedded Framework but Microsoft has started pushing the Edge embedder framework downstream.  Better that than a 50meg DLL.

Just a thought.  I feel like TGUI really revitalized TG in may respects and using the small React (Inferno) like platform made it easy to make modules for it.  I want to ask first before I start burning out my brain putting in a pull.

**Comment by moonheart08** [comment, marked as answer]
SS14 development seems very very hard against webtech in the game, and I personally am too.
The current UI system has vastly better performance than HTML+CSS+JS.

To add, WebView2 is not cross platform, and SS14 supports Windows, MacOS, and Linux, in addition to much of the development team using Linux. It's a non-option even if web embedding was considered ok for SS14.

**Comment by gradientvera** [comment]
No. We have banned the use of any web embedding whatsoever in vanilla SS14.

**Comment by PJB3005** [comment]
Webview2 is off the table due to lack of cross plat support and no off-screen rendering. Otherwise CEF is also off the table because it's absolutely terrible as a UI system for a ton of reasons ranging from performance to integration.

SS14 will never use web stuff for interfaces, period.
